src/Raven.Client/Documents/Commands/Batches/DeleteCommandData.cs

```diff
@@ -9,24 +9,38 @@ namespace Raven.Client.Documents.Commands.Batches
     public class DeleteCommandData : ICommandData
     {
         public DeleteCommandData(string id, string changeVector)
+            :this(id, changeVector, changeVector)
+        {
+            
+        }
+        public DeleteCommandData(string id, string changeVector, string oldChangeVector)
         {
             Id = id ?? throw new ArgumentNullException(nameof(id));
             ChangeVector = changeVector;
+            OldChangeVector = oldChangeVector;
         }
 
+        public string OldChangeVector { get; }
         public string Id { get; }
         public string Name { get; } = null;
         public string ChangeVector { get; }
         public CommandType Type { get; } = CommandType.DELETE;
+        public BlittableJsonReaderObject Document { get; set; }
 
         public virtual DynamicJsonValue ToJson(DocumentConventions conventions, JsonOperationContext context)
         {
-            return new DynamicJsonValue
+            var json = new DynamicJsonValue
             {
                 [nameof(Id)] = Id,
                 [nameof(ChangeVector)] = ChangeVector,
-                [nameof(Type)] = Type.ToString()
+                [nameof(Type)] = Type.ToString(),
+                [nameof(Document)] = Document
```

Comment:
Can we add it to the json only if it's not null? Most of the time it's going to be null so no need to send additional property over the wire